### ISSUE TITLE
Viewport camera distance problem [$15]

### ISSUE THREAD

**Issue by robnee** [openscad/openscad#916]
I mostly design objects in units of inches.  The scale of the finished objects is usually on the order of an inch or two overall.  When zooming the view using perspective mode once the camera distance is closer than about 10 - 20 "units" the camera starts to fly into the interior of the model.  This makes zooming in on fine details impossible as I can't visualize the model any closer than 10-20 inches away.  If I change my model to specify units in mm this problem goes away because now I can now get as close as 10-20mm from the object surface before I fly into the interior.  For various reasons using inches is necessary.

Here is a trivial bit of sample code to aid demonstration:

```
// h = 1.000 / 39.4; // meter
// h = 1.000; // inch
h = 1.000 * 25.4; // mm

union () {
    cube ([h, h, h]);

    translate ([-h/4, -h/4, 0])
        cylinder (h = h, d =h, $fn = 30);
}
```

Render the model in inch and then in mm and attempt to zoom into the joint between the cylinder and cube to see the difference.  The mm model will allow you to view up close before piercing the skin of the model with the camera.

If someone was to specify a model in feet or meters the effect becomes much worse.  It becomes impossible to view the model at anything closer than a "thousand foot view".

I would like to propose at special variable, perhaps $vd, that would scale the view distance so that models specified in large units such as inch, foot and meter can be rendered for viewing properly. 

I understand that I can render the model with an enclosing "scale (100)" but I'm not convinced that is really the best solution.  It must be commented out when rendering for export.  It seems that the units of the model should not affect the ability of openSCAD to visualize it. 
##

**Comment by thehans** [comment]
I can't reproduce any issue with this sample code.  Seems to have been fixed long ago, 2015.03-2 zooms in fine at "meter" scale.  

The near clipping plane in perspective mode is scaled based on the zoom distance:
https://github.com/openscad/openscad/blob/5ccfd7b7da8cb5b0143946ae4b2893de5829bac5/src/GLView.cc#L98

Its been that way for a while it seems.

**Comment by thehans** [comment]
After entirely too much code archaeology, I've determined that this was fixed in 458d60faa6287234df811d5f2a50fd45a9df13d1 on Jun 19 2014 by kintel.

@kintel please claim your bounty :smiley:
